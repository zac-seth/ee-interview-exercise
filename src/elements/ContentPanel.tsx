import { FC } from 'react';

import { Panel, PanelVariant } from './Panel.style';

export type ContentPanelProps = {
  children: JSX.Element;
};

export const ContentPanel: FC<ContentPanelProps> = ({ children }) => (
  <Panel variant={PanelVariant.CONTENT} data-testid='content-panel'>
    {children}
  </Panel>
);
