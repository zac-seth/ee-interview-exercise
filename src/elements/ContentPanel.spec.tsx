import { render, screen } from '@testing-library/react';

import { ContentPanel } from './ContentPanel';

describe(`ContentPanel`, () => {
  it(`Should render a content panel`, () => {
    render(
      <ContentPanel>
        <div>Content in a panel!</div>
      </ContentPanel>
    );

    expect(screen.getByTestId(`content-panel`)).toMatchSnapshot();
  });

  it(`Should choose the content panel background color style`, () => {
    render(
      <ContentPanel>
        <div>Content in a panel!</div>
      </ContentPanel>
    );

    const contentPanel = screen.getByTestId(`content-panel`);

    expect(contentPanel).toHaveStyleRule(`background-color`, `#FFF`);
  });
});
