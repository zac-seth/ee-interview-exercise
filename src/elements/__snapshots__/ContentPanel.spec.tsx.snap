// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPanel Should render a content panel 1`] = `
.c0 {
  width: 33.25rem;
  padding: .25rem;
  border-radius: .25rem;
  background-color: #FFF;
}

<section
  class="c0"
  data-testid="content-panel"
>
  <div>
    Content in a panel!
  </div>
</section>
`;
